### PR TITLE
Migrate settings edit birthday page to authflow v2

### PR DESCRIPTION
**Pull request by Kuixz**
This PR will likely have merge conflicts with #2 in `translations.json` and a few other places.

### CODE REVIEW

File: resources/authgear/templates/en/web/authflowv2/__date_input.html [hunk 1]
```diff
@@ -0,0 +1,23 @@
...
+type="date"
+class="date-input__trigger" 
+name="{{ .Name }}"
+value="{{ .Value }}"
+data-date-input-target="input" 
+data-action="input->date-input#updateLabel"
+{{ if .Min }} max="{{ .Max }}" {{ end }}
+{{ if .Max }} max="{{ .Max }}" {{ end }}>
+
+{{ end }}
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/__date_input.html]: Add a newline at the end of this file

File: resources/authgear/templates/en/web/authflowv2/__date_input.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+
+{{ define "authflowv2/__date_input.html" }}
+
+<input 
+type="date"
+class="date-input__trigger" 
+name="{{ .Name }}"
+value="{{ .Value }}"
+data-date-input-target="input" 
+data-action="input->date-input#updateLabel"
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/__date_input.html]: Remove these 2 lines.

File: resources/authgear/templates/en/web/authflowv2/__date_input.html [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+    "Max" .Today
+  )
+}}
+-->
+
+{{ define "authflowv2/__date_input.html" }}
+
+<input 
+type="date"
+class="date-input__trigger"
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/__date_input.html]: Rename this class to date-input

Comment (cychiuae) [reply resources/authgear/templates/en/web/authflowv2/__date_input.html]: I think we can just use the existing `input` class and `date-input` is just to style the little calendar icon

File: resources/authgear/templates/en/web/authflowv2/settings_profile_edit_birthdate.html [hunk 4]
```diff
@@ -0,0 +1,40 @@
...
+  "BackHref" (call $.MakeURL "/settings/profile")
+  "Title" (translate "v2.page.settings-profile-edit-birthdate.default.navbar-title" nil)
+)
+}}
+
+<form
+data-controller="turbo-form"
+data-action="submit->turbo-form#submitForm"
+class="settings-profile-editor gap-4"
+method="post">
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/settings_profile_edit_birthdate.html]: Identation

File: resources/authgear/templates/en/web/authflowv2/settings_profile_edit_birthdate.html [hunk 5]
```diff
@@ -0,0 +1,40 @@
+{{ template "authflowv2/__html_head.html" . }}
```
Comment (cychiuae) [inline resources/authgear/templates/en/web/authflowv2/settings_profile_edit_birthdate.html]: We don't need this. It should be included in `authflowv2/__settings_page_frame.html`

File: authui/src/authflowv2/components/date-input.css [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+    &:focus {
+      @apply ring-inset
+          ring-[length:var(--date-input\_\_trigger-ring-width--focus)]
+          ring-[color:var(--date-input\_\_trigger-ring-color--focus)];
+    }
+
+    &::-webkit-calendar-picker-indicator {
+      height: 1.5rem;
+      width: 1.5rem;
+      /* cursor: pointer; */
```
Comment (cychiuae) [inline authui/src/authflowv2/components/date-input.css]: Remove it if it is not necessary

File: resources/authgear/templates/en/web/authflowv2/settings_profile_edit_birthdate.html [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+{{ template "authflowv2/__settings_page_frame.html" . }}
+
+{{ define "page-content" }}
+
+{{ template "authflowv2/__navbar.html"
+(dict
+  "BackHref" (call $.MakeURL "/settings/profile")
```
Comment (yyorkchan) [inline resources/authgear/templates/en/web/authflowv2/settings_profile_edit_birthdate.html]: Thinking if BackTitle (The word "Back") should be included for desktop view @cychiuae

**Comment by Kuixz** [comment]
Addressed all of @cychiuae changes ✓✓✓✓✓✓

@yyorkchan to be precise, what action / change are you suggesting?

**Pull request merged**